lib/kube/proxy/websocket_client_test.go

```diff
@@ -241,7 +323,111 @@ func (e *wsStreamExecutor) stream(conn *gwebsocket.Conn, options clientremotecom
 	return err
 }
 
-func (e *wsStreamExecutor) connectViaWebsocket() error {
+const (
+	// portforwardDataChan is the websocket channel for transferring data.
+	portforwardDataChan = iota
+	// portforwardErrChan is the websocket channel for transferring errors.
+	portforwardErrChan
+)
+
+// portforward opens a listener at the specified port and waits for new connections.
+// once a connection is received, it waits for the upstream data and writes it
+// into the original connection.
+// Due to portforward websocket limitations, the listener do not accept
+// concurrent connections.
+func (e *wsStreamClient) portforward(remoteConn *gwebsocket.Conn) (err error) {
+	if e.localPort == nil || e.readyChan == nil {
+		return trace.Errorf("cannot use portforward without proper initialization")
```

Comment:
I've been told before to avoid trace.Errorf and default to trace.BadParameter instead:

```suggestion
		return trace.BadParameter("cannot use portforward without proper initialization")
```